Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -142,7 +160,10 @@ ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertion(ChildChan
                 hadElementChild = true;
 
             removeBetween(nullptr, child->protectedNextSibling().get(), *child);
-            auto subtreeObservability = notifyChildNodeRemoved(*this, *child);
+            auto [subTreeSize, subtreeObservability, subtreeChildrenDeletionCanBeDelayed] = notifyChildNodeRemoved(*this, *child);
+            treeSize += subTreeSize;
+            if (subtreeChildrenDeletionCanBeDelayed == ChildrenDeletionCanBeDelayed::No)
```

Comment:
We should probably assert that subtreeChildrenDeletionCanBeDelayed isn't unknown here.